numba/np/arrayobj.py

```diff
@@ -266,6 +267,29 @@ def update_array_info(aryty, array):
                                           get_itemsize(context, aryty))
 
 
+def normalize_axis(func_name, ndim, axis):
+    """Constrain axis values to valid positive values."""
+    raise NotImplementedError()
+
+
+@overload(normalize_axis)
+def _normalize_axis_overloads(func_name, ndim, axis):
+    if not isinstance(func_name, Literal):
+        raise errors.TypingError("func_name must be a str literal.")
+
+    msg = f"{func_name.literal_value}: Axis out of bounds"
```

Comment:
Instead of an overload, you can either use `@generated_jit` or `@register_jitable`

Reply:
I switched to `generated_jit`. I don't see how I could use `register_jitable` though, because I think the error message needs to be known at compile time, and register_jitable can't dispatch on literal types, can it?

Reply:
That's correct! But it should be possible after https://github.com/numba/numba/pull/8134 gets merged.